mobile/src/main/java/org/openhab/habdroid/background/AlarmChangedWorker.java

```diff
@@ -0,0 +1,101 @@
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ITEM;
+
+public class AlarmChangedWorker extends Worker {
+    private static final String TAG = AlarmChangedWorker.class.getSimpleName();
+
+    public AlarmChangedWorker(
+            @NonNull Context context,
+            @NonNull WorkerParameters params) {
+        super(context, params);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    @Override
+    public Result doWork() {
+        Context context = getApplicationContext();
+        NotificationManager nm =
+                (NotificationManager) context.getSystemService(NOTIFICATION_SERVICE);
+
+        ConnectionFactory.waitForInitialization();
+
+        Connection connection;
+        ConnectionException failureReason;
+        try {
+            Log.d(TAG, "Trying to get connection");
+            connection = ConnectionFactory.getUsableConnection();
+            failureReason = null;
+        } catch (ConnectionException e) {
+            Log.e(TAG, "ConnectionException", e);
+            connection = null;
+            failureReason = e;
+        }
+
+        if (connection == null) {
+            Log.e(TAG, "Got no connection " + failureReason);
+            return BackgroundUtils.retryOrFail(
+                    context.getString(R.string.error_sending_alarm_clock_no_connection),
+                    context.getString(R.string.error_sending_alarm_clock_retry),
+                    getRunAttemptCount(),
+                    BackgroundUtils.makeRetryAction(context, NOTIFICATION_ID_SEND_ALARM_CLOCK),
+                    NOTIFICATION_ID_SEND_ALARM_CLOCK,
+                    context, nm);
+        }
+
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        String nextAlarm;
+        AlarmManager.AlarmClockInfo alarmClockInfo = alarmManager.getNextAlarmClock();
+        if (alarmClockInfo == null) {
+            nextAlarm = "0";
+        } else {
+            nextAlarm = String.valueOf(alarmClockInfo.getTriggerTime());
+        }
+        String item = PreferenceManager.getDefaultSharedPreferences(context)
+                .getString(PREFERENCE_ALARM_CLOCK_ITEM, "");
+
+        String url = String.format(Locale.US, "rest/items/%s", item);
+        SyncHttpClient.HttpResult result = connection.getSyncHttpClient()
+                .post(url, nextAlarm, "text/plain;charset=UTF-8");
+
+        if (result.isSuccessful()) {
+            Log.d(TAG, "Alarm clock successfully sent: " + nextAlarm);
+            nm.cancel(NOTIFICATION_TAG_BACKGROUND, NOTIFICATION_ID_SEND_ALARM_CLOCK);
+            return Result.success();
+        } else {
+            Log.e(TAG, "Error sending alarm clock. Got HTTP error "
+                    + result.statusCode, result.error);
+
+            return BackgroundUtils.retryOrFail(
```

Comment:
The device could be connected to a wifi that has a webserver at the same ip/dns than the openHAB server in the home wifi. The app connects to this server, but gets a 404, because the rest api isn't present at all.